jerry-core/ecma/base/ecma-alloc.c

```diff
@@ -80,12 +80,37 @@ JERRY_STATIC_ASSERT (sizeof (ecma_extended_object_t) - sizeof (ecma_object_t) <=
   ALLOC (ecma_type) \
   DEALLOC (ecma_type)
 
-DECLARE_ROUTINES_FOR (object)
 DECLARE_ROUTINES_FOR (number)
 DECLARE_ROUTINES_FOR (collection_header)
 DECLARE_ROUTINES_FOR (collection_chunk)
-DECLARE_ROUTINES_FOR (string)
-DECLARE_ROUTINES_FOR (getter_setter_pointers)
+
+/**
+ * Allocate memory for ecma-object
+ *
+ * @return pointer to allocated memory
+ */
+inline ecma_object_t * __attr_always_inline___
+ecma_alloc_object (void)
```

Comment:
I would not allow any macro generated functions in the future. So bad for debugging, we are not optimizing for source code size, and it does not affect binary size at all. The existing macro is legacy, and could be removed, but not in this patch. A low hanging fruit for somebody.